### PR TITLE
docs: DOC-419 architecture section

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
This PR is a breakout of PR #1009 . The architecture and troubleshooting changes were extracted out into this PR.

This PR restructures the information layout in the main menu (navbar).  

💻 [Preview Link](https://deploy-preview-1026--docs-spectrocloud.netlify.app/)

## Architecture

A section dedicated to the architecture and technical details of the product is missing (excluding security). This section has been added along with elements from the troubleshooting section that better fit an architecture topic. 

CleanShot 2022-12-01 at 09 57 25

## Troubleshooting
As mentioned in the Architecture section, this section was cleaned up and now only focuses on troubleshooting steps.  The existing content has been broken up into their respective category and file..

CleanShot 2022-12-01 at 09 56 18

The content of the index page.
CleanShot 2022-12-01 at 10 01 38

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1026#" title="Author self-approved">karl-cardenas-coding</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/14-troubleshooting/02-kubernetes-tips.md
Comment (ritawatson) [inline content/docs/14-troubleshooting/02-kubernetes-tips.md:20]:
```suggestion
The table below lists useful Kubernetes log types that can aid you in the debugging process. The [Kubernetes Logging Architecture](https://kubernetes.io/docs/concepts/cluster-administration/logging/) page is a good resource you can review to help gain a better understanding of the logging architecture.  
```

File: content/docs/14-troubleshooting/04-cluster-deployment.md
Comment (ritawatson) [inline content/docs/14-troubleshooting/04-cluster-deployment.md:144]:
```suggestion
Installation of the Gateway cluster may run into errors or get stuck in the provisioning state for various reasons like lack of infrastructure resources, lack of availability of IP addresses, or inability to perform NTP sync. 
```

File: content/docs/14-troubleshooting/04-cluster-deployment.md
Comment (ritawatson) [inline content/docs/14-troubleshooting/04-cluster-deployment.md:148]:
```suggestion
Intermittent errors are displayed on the **Cluster Details** page next to the relevant orchestration task. The **Events** tab on this page also provides helpful insights into lower-level operations currently being performed. Suppose you believe the orchestration is stuck or failed due to an invalid selection of infrastructure resources or an intermittent problem with the infrastructure. You may reset the Gateway by clicking on the **Reset** button on the Gateway widget. The Gateway state will transition to Pending. A Gateway in the Pending state allows you to reconfigure the Gateway and start provisioning a new Gateway cluster. If the problem persists, contact Spectro support via the Service Desk or our [support page](http://support.spectrocloud.io/).
```

File: content/docs/14-troubleshooting/06-nodes.md
Comment (ritawatson) [inline content/docs/14-troubleshooting/06-nodes.md:18]:
```suggestion
Palette performs a rolling upgrade on the nodes when a change is detected in the kubeadm config. Below are some actions that can cause the kubeadm configuration to change and result in nodes getting upgraded:
```

**Review by ritawatson — APPROVED** [review]
LGTM with some comments:
- Search for _will_ and make it active voice.
- Avoid e.g., etc., and i.e. (If you agree, I can add these to our Vale reject folder)
- Voice: Address the user with _you_ throughout instead of _the user_ or _the customer_.
- Is it NameSpace or namespace? Also, add hyphens in namespace-to-pod mapping: _The following table gives the namespace to pod mapping for Palette Tenant Cluster._
Saw it in the preview but not in changed files.

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1026#" title="Author self-approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1026#pullrequestreview-1209231655" title="Approved">ritawatson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
